pylds/hamiltonians.py

```diff
@@ -0,0 +1,259 @@
+        Points in phase space.
+
+    Returns
+    -------
+    H : array_like, shape(n,)
+        Hamiltonian at points u, in phase space at time t
+
+    """
+    points_positions = u.T[:2]
+    points_momenta = u.T[2:4]
+    x, y = points_positions
+    p_x, p_y = points_momenta
+    # Potential energy function
+    H = 0.5*(x**2 + y**2) + (y * x**2) - (1/3)*y**3 + 0.5*(p_x**2 + p_y**2)
+    return H
+
+def HenonHeiles_potential(positions, PARAMETERS = None):
+    """
+    Potential energy function for 2DoF Henon-Heiles system.
+    Functional form: V = 1/2(x**2 + y**2) + yx**2 - y**3/3, with positions = (x, y).
+
+    Parameters
+    ----------
+    positions : array_like, shape(n,)
+        Points in phase space.
+
+    Returns
+    -------
+    V : array_like, shape(n,)
+        Potential energy at points u.
+
+    """
+    x, y = positions.T
+    V = (1/2)*(x**2 + y**2) + (y * x**2) - (1/3)*y**3
+    return V
+
+def NFSaddle_Hamiltonian(t, u):
+    """
+    Hamiltonian for a 2DoF Saddle.
+    Functional form: H = 1/2(p_x**2 + p_y**2) + 1/2(y**2 - x**2), with u = (x, y, p_x, p_y).
+
+    Parameters
+    ----------
+    t : float
+        Time. (This Hamiltonian is independent of time.)
+
+    u : array_like, shape(n,)
+        Points in phase space.
+
+    Returns
+    -------
+    H : array_like, shape(n,)
+        Hamiltonian at points u, in phase space at time t
+
+    """
+    points_positions = u.T[:2]
+    points_momenta = u.T[2:4]
+    x, y = points_positions
+    p_x, p_y = points_momenta
+    H = 0.5*(p_x**2 + p_y**2 + y**2 - x**2)
+    return H
+
+def NFSaddle_potential(positions, PARAMETERS = None):
+    """
+    Potential energy function for a 2DoF Saddle.
+    Functional form: V = 1/2(y**2 - x**2), with positions = (x, y).
+    ----------
+    positions : array_like, shape(n,)
+        Points in phase space.
+
+    Returns
+    -------
+    V : array_like, shape(n,)
+        Potential energy at points u.
+    """
+    x, y = positions.T
+    V = 0.5*(y*y - x*x)
+    return V
+
+def kinetic_squares(t,u):
+    """
+    Kinetic energy of the form sum of squares.
+
+    Parameters
+    ----------
+    t : float
+        Time. (This Hamiltonian is independent of time.)
+
+    u : array_like, shape(n,)
+        Points in phase space.
+
+    Returns
+    -------
+    T : array_like, shape(n,)
+        Kinetic energy at points u, in phase space at time t
+
+    """
+    N_dim = np.floor(u.shape[1]/2).astype('int')
+    points_momenta = u.T[N_dim:2*N_dim]
+    return 0.5* np.sum(points_momenta**2, axis=0)
+
+def Hamiltonian_from_potential(potential):
+    """
+    Forms a Hamiltonian by combining input potential function with kinetic energy 'kinetic_squares'.
+
+    Parameters
+    ----------
+    potential : function
+
+    Returns
+    -------
+    Hamiltonian : function
+        Sum of 'potential' and 'kinetic_squares'.
+
+    """
+    def Hamiltonian(t,u):
+        return potential(u) + kinetic_squares(t,u)
+    return Hamiltonian
```

Comment:
WRONG INPUT VARIABLE for `potential`.
Here, `u` is taken as an array of phase space points. The function `kinetic_squares` correctly takes points corresponding to conjugate momenta axes (`momenta_points`) from `u`, but all potential energy functions only accept `positions`, leading to a `ValueError` when running `NFSaddle` example.

Reply:
Thanks for finding this issue!

Reply:
Fixed